### ISSUE TITLE
[1.18.2] `ArrayIndexOutOfBoundsException: Index -12582613 out of bounds for length 315`

### ISSUE THREAD

**Issue by SettingDust** [Silverminer007/Shrines#54]
After looking into the issue. The order of args here is wrong.
https://github.com/Silverminer007/Shrines/blob/1.18.2+-4.x.x/src/main/java/com/silverminer/shrines/structures/spawn_criteria/HeightSpawnCriteria.java#L57-L58
`getCornerHeights` is `int x0, int z0, int x1, int z1` but the passing args are `x0 x1 z0 z1`
```
java.lang.ArrayIndexOutOfBoundsException: Index -12582613 out of bounds for length 315
	at net.minecraft.world.level.levelgen.Aquifer$NoiseBasedAquifer.m_188445_(Aquifer.java:489) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B}
	at net.minecraft.world.level.levelgen.Aquifer$NoiseBasedAquifer.m_207104_(Aquifer.java:284) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B}
	at net.minecraft.world.level.levelgen.NoiseChunk.m_209215_(NoiseChunk.java:204) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B}
	at net.minecraft.world.level.levelgen.material.MaterialRuleList.m_207387_(MaterialRuleList.java:15) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:classloading,pl:rei_plugin_compatibilities:B}
	at net.minecraft.world.level.levelgen.NoiseChunk.m_209247_(NoiseChunk.java:236) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B}
	at net.minecraft.world.level.levelgen.NoiseBasedChunkGenerator.m_158413_(NoiseBasedChunkGenerator.java:255) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:computing_frames,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B}
	at net.minecraft.world.level.levelgen.NoiseBasedChunkGenerator.m_142647_(NoiseBasedChunkGenerator.java:172) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:computing_frames,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B}
	at net.minecraft.world.level.chunk.ChunkGenerator.m_156179_(ChunkGenerator.java:723) ~[server-1.18.2-20220404.173914-srg.jar%23274!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,re:classloading,pl:accesstransformer:B,pl:rei_plugin_compatibilities:B,pl:mixin:APP:modernfix-common.mixins.json:perf.cache_strongholds.ChunkGeneratorMixin,pl:mixin:APP:betterfortresses_forge.mixins.json:DisableVanillaFortressesMixinForge,pl:mixin:APP:structure_gel.mixins.json:ChunkGeneratorMixin,pl:mixin:APP:citadel.mixins.json:ChunkGeneratorMixin,pl:mixin:APP:bettermineshafts_forge.mixins.json:DisableVanillaMineshaftsMixinForge,pl:mixin:APP:integrated_stronghold.mixins.json:DisableVanillaStrongholdsMixin,pl:mixin:APP:betterwitchhuts_forge.mixins.json:DisableVanillaWitchHutsMixinForge,pl:mixin:APP:betteroceanmonuments_forge.mixins.json:DisableVanillaMonumentsMixinForge,pl:mixin:APP:betterstrongholds_forge.mixins.json:CancelSearchVanillaStrongholdMixinForge,pl:mixin:APP:betterstrongholds_forge.mixins.json:DisableVanillaStrongholdsMixinForge,pl:mixin:APP:mixins.shrines.json:MixinChunkGenerator,pl:mixin:APP:betterdeserttemples_forge.mixins.json:DisableVanillaPyramidsMixinForge,pl:mixin:A}
	at com.silverminer.shrines.structures.spawn_criteria.HeightSpawnCriteria.getCornerHeights(HeightSpawnCriteria.java:65) ~[Shrines-1.18.2-4.1.0.jar%23225!/:1.18.2-4.1.0] {re:classloading,pl:rei_plugin_compatibilities:B}
	at com.silverminer.shrines.structures.spawn_criteria.HeightSpawnCriteria.getLowestY(HeightSpawnCriteria.java:57) ~[Shrines-1.18.2-4.1.0.jar%23225!/:1.18.2-4.1.0] {re:classloading,pl:rei_plugin_compatibilities:B}
	at com.silverminer.shrines.structures.spawn_criteria.HeightSpawnCriteria.test(HeightSpawnCriteria.java:50) ~[Shrines-1.18.2-4.1.0.jar%23225!/:1.18.2-4.1.0] {re:classloading,pl:rei_plugin_compatibilities:B}
	at com.silverminer.shrines.structures.ShrinesStructure.lambda$place$0(ShrinesStructure.java:33) ~[Shrines-1.18.2-4.1.0.jar%23225!/:1.18.2-4.1.0] {re:classloading,pl:rei_plugin_compatibilities:B}
	at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:?] {}
```

Original: Dawncraft Discord https://discord.com/channels/809137878646521896/1165127730241736704/1165127730241736704

**Comment by SettingDust** [comment]
Should be fixed with https://legacy.curseforge.com/minecraft/mc-mods/mod-compat-fixes/files/4810969

https://github.com/SettingDust/dawncraft-fixes/blob/main/src/main/java/io/github/settingdust/dawncraftfixes/mixin/shrines/MixinHeightSpawnCriteria.java

**Comment by SettingDust** [comment]
bump
